BritishGeologicalSurvey/etlhelper
# Sqlserver tests

### Description

This merge request adds integration tests for MS SQL server.  These tests require a connection to a MS SQL database and are mainly run by BGS developers and internal CI processes.

The integration tests for MSSQL, Oracle and SQLite were modified to be as similar as possible.  In general, if you can run `copy_rows` then most things are working.  For this reason, a number of pre-existing MS SQL tests have been removed.  They are still present within the commit history in case we need them (removed by this commit - https://github.com/BritishGeologicalSurvey/etlhelper/commit/77930546923becdf2f936afdc96a8b5fa9eee1c8).

Closes #6.  In the time since `etlhelper` has been released, the upstream issue with ODBC drivers on Debian has been resolved.

### To test

+ [x] Run integration tests locally
+ [x] Run integration tests locally via `bin/run_tests_for_developers.sh`, which confirms Docker build.

Note: these tests run against a different server to the original BGS ETL tests.  See GitLab CI variables for new connection details.  Also, as we now have a default `etlhelper` schema, it is no longer necessary to specify the schema name in the queries.
